core/lib/libtomcrypt/ecc.c

```diff
@@ -379,3 +346,55 @@ TEE_Result crypto_acipher_ecc_shared_secret(struct ecc_keypair *private_key,
 	ecc_free(&ltc_public_key);
 	return res;
 }
+
+static const struct crypto_ecc_keypair_ops ecc_keypair_ops = {
+	.generate = &ltc_ecc_generate_keypair,
+	.sign = &ltc_ecc_sign,
+	.shared_secret = &ltc_ecc_shared_secret,
+};
+
+static const struct crypto_ecc_public_ops ecc_public_key_ops = {
+	.free = &ltc_ecc_free_public_key,
+	.verify = &ltc_ecc_verify,
+};
+
+TEE_Result crypto_asym_alloc_ecc_keypair(struct ecc_keypair *s,
```

Comment:
The two implementations of this function are quite similar, with the key difference with the `ops` assignment.
It is tempting to move this into the common ` crypto_acipher_alloc_ecc_keypair()` and instead let the different implementation make the `ops` struct available.

Reply:
@cneveux, comments?

Reply:
@jenswi-linaro,
Not sure to understand your though at this point. Did I miss something?

I think it's better to have functions in each driver to allow the key as the driver expect to get if it support it and then let it to set the ops function of driver function implemented (ecc keys are also used for SM2 which is pretty new Chinese algo not supported by all drivers).
In future as we already discussed it will be good to let driver to use or not bignumber for the key buffers. It can prevent back and forth buffer conversion when not necessary.

Concerning the `ops` field, I'll change it to be no more a `void *` and define:
```
struct crypto_ecc_public_ops{
   TEE_Result verify(...);
   TEE_Result encrypt(...);
   void free(...);
}
```
and
```
struct crypto_ecc_keypair_ops{
    TEE_Result generate(...);
    TEE_Result sign(...);
    TEE_Result shared_secret(...);
    TEE_Result decrypt(...);
    TEE_Result derive(...);
}
```

Reply:
My comment was only for the software implementations, after `drvcrypt_asym_alloc_ecc_keypair()` has returned `TEE_ERROR_NOT_IMPLEMENTED`. I expect that the software implementations should be able to use the bignum implementation available, if not then we need to do a completely new take on bignums.

Reply:
OK, I see.
But to say yes or not to your proposal I need to investigate libtomcrypt to see why doing a (not optimized) `bn_alloc_max` allocation to each key fields instead of doing just a `crypto_bignum_allocate(key_size_bits)` as done in mbedTLS.
If we want to have tomcrypt and mbedTLS working without problem with a common key allocation, we have to make sure that both SW library tomcrypt and mbedTLS are compatible in term of key allocation.

Reply:
I'm sorry, I missed the difference in allocation. Let's take that later on then.

Reply:
ok, thanks